### PR TITLE
clean up documentation

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
Removing some of the unsupported flags after adding new BC interface

### CODE REVIEW

File: Docs/source/usage/parameters.rst [hunk 1]
```diff
@@ -1584,54 +1618,6 @@ Numerics and algorithms
...
-
-* ``warpx.do_pml_Hi`` (`2 floats in 2D`, `3 floats in 3D`; default: `1 1 1`)
-    The directions along which one wants a pml boundary condition for upper boundaries on mother grid.
-
-* ``warpx.do_silver_mueller`` (`0` or `1`; default: 1)
-    Whether to use the Silver-Mueller absorbing boundary conditions. These
-    boundary conditions are simpler and less computationally expensive than
-    the PML, but are also less effective at absorbing the field. They only
-    work with the Yee Maxwell solver.
-
```
Comment (RevathiJambunathan) [inline Docs/source/usage/parameters.rst:1634]: these flags are not required anymore.

File: Docs/source/usage/parameters.rst [hunk 2]
```diff
@@ -1584,54 +1618,6 @@ Numerics and algorithms
...
-    Whether to use divergence cleaning for E in the PML region.
-    The value must match ``warpx.do_pml_divb_cleaning`` (either both false or both true).
-    This option seems to be necessary in order to avoid strong Nyquist instabilities in 3D simulations with the PSATD solver, open boundary conditions and PML in all directions. 2D simulations and 3D simulations with open boundary conditions and PML only in one direction might run well even without divergence cleaning.
-    This option is implemented only for the PSATD solver.
-
-* ``warpx.do_pml_divb_cleaning`` (`bool`; default: 1)
-    Whether to use divergence cleaning for B in the PML region.
-    The value must match ``warpx.do_pml_dive_cleaning`` (either both false or both true).
-    This option seems to be necessary in order to avoid strong Nyquist instabilities in 3D simulations with the PSATD solver, open boundary conditions and PML in all directions. 2D simulations and 3D simulations with open boundary conditions and PML only in one direction might run well even without divergence cleaning.
-    This option is implemented only for the PSATD solver.
```
Comment (RevathiJambunathan) [inline Docs/source/usage/parameters.rst:1621]: moved these further up in the doc -- right after all BCs are listed, with a sub-title "additional pml parameters"

**Review by EZoni — APPROVED** [review]
Looks good to me, thank you!

**Review by ax3l — APPROVED** [review]
Thank you! :tada:

File: Source/WarpX.H [hunk 3]
```diff
@@ -365,7 +365,6 @@ public:
     void EvolveE (int lev, PatchType patch_type, amrex::Real dt);
     void EvolveF (int lev, PatchType patch_type, amrex::Real dt, DtType dt_type);
     void EvolveG (int lev, PatchType patch_type, amrex::Real dt, DtType dt_type);
-    void ApplySilverMuellerBoundary (amrex::Real dt);
```
Comment (ax3l) [inline Source/WarpX.H:368]: Unused declaration :heavy_check_mark:

**Pull request merged**